### ISSUE TITLE
@JsonIgnore in 2.3.6 breaks object reference

### ISSUE THREAD

**Issue by auke-** [immutables/immutables#469]
We use the following pattern on several occasions, which works fine until Immutables 2.3.6:
```
@JsonSerialize(as = ImmutableA.class)
@JsonDeserialize(as = ImmutableA.class)
@Value.Immutable
public interface A {
    @JsonIdentityReference(alwaysAsId = true)
    @JsonIdentityInfo(
            resolver = BInstanceResolver.class,
            generator = ObjectIdGenerators.PropertyGenerator.class,
            property = "id")
    B getB();
}

@JsonSerialize(as = ImmutableB.class)
@JsonDeserialize(as = ImmutableB.class)
@Value.Immutable
public interface B {
    String getId();
}
```

Probably due to the @JsonIgnore annotation on the 'id' member of ImmutableB:
```
java.lang.IllegalArgumentException: Invalid Object Id definition for <package>.ImmutableB: can not find property with name 'id'
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.createContextual(BeanDeserializerBase.java:704)
	at com.fasterxml.jackson.databind.DeserializationContext.handleSecondaryContextualization(DeserializationContext.java:681)
	at com.fasterxml.jackson.databind.DeserializationContext.findContextualValueDeserializer(DeserializationContext.java:445)
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer.findDeserializer(StdDeserializer.java:965)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.resolve(BeanDeserializerBase.java:501)
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCache2(DeserializerCache.java:293)
	at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCacheValueDeserializer(DeserializerCache.java:244)
	at com.fasterxml.jackson.databind.deser.DeserializerCache.findValueDeserializer(DeserializerCache.java:142)
	at com.fasterxml.jackson.databind.DeserializationContext.findRootValueDeserializer(DeserializationContext.java:476)
	at com.fasterxml.jackson.databind.ObjectMapper._findRootDeserializer(ObjectMapper.java:3899)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:3794)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2842)
```

**Comment by elucash** [comment]
`@JsonIgnore` is on the field, we have an accessor generated with `@JsonProperty("id")`. It can be a Jackson bug. Can you, please, store and edit generated class: remove `@Value.Immutable` to stop regenerating and manually remove `@JsonIgnore` from the field to confirm that it works without it. We might put this auto-ignoring under a feature flag to generate code as before. Thank you for reporting this!

**Comment by elucash** [comment]
I've checked, this functionality is indeed broken by the `JsonIngnore` annotation, but it shouldn't break it.
I was able to reproduce the problem with Immutables, then I've tried the same approach with pojos and was unable to reproduce, crazy stuff

```java
// this one works despite @JsonIgnore
class Pojos {
	static class AsIs {
		@JsonIgnore
		private ById by;
		@JsonIdentityReference(alwaysAsId = true)
		@JsonIdentityInfo(
				resolver = InstanceResolver.class,
				generator = ObjectIdGenerators.PropertyGenerator.class,
				property = "id")
		@JsonProperty("by")
		ById getBy() {
			return by;
		}
		// @JsonProperty("by")
		void setBy(ById by) {
			this.by = by;
		}
	}
	static class ById {
		@JsonIgnore
		private String id;
		@JsonProperty("id")
		String getId() {
			return id;
		}
		// @JsonProperty("id")
		void setId(String id) {
			this.id = id;
		}
	}

	static class InstanceResolver extends SimpleObjectIdResolver {
		@Override
		public ObjectIdResolver newForDeserialization(Object context) {
			return new InstanceResolver();
		}
		@Override
		public Object resolveId(IdKey id) {
			ById byId = new ById();
			byId.setId(id.key.toString());
			return byId;
		}
	}

	public static void main(String... args) throws Exception {
		ById byId = new ById();
		byId.setId("abc");
		AsIs is = new AsIs();
		is.setBy(byId);

		ObjectMapper mapper = new ObjectMapper();
		String json = mapper.writeValueAsString(is);
		System.out.println(json);

		AsIs is2 = mapper.readValue(json, AsIs.class);
		System.out.println(is2);
	}
}
```

**Comment by auke-** [comment]
For what it's worth, the generated code for ImmutableB is:
```
@SuppressWarnings({"all"})
@ParametersAreNonnullByDefault
@Generated({"Immutables.generator", "B"})
@Immutable
public final class ImmutableB implements B {
  @JsonIgnore
  private final String id;

  private ImmutableB(String id) {
    this.id = id;
  }

  /**
   * @return The value of the {@code id} attribute
   */
  @JsonProperty("id")
  @Override
  public String getId() {
    return id;
  }

  ...
}
```

It does work when the @JsonIgnore annotation is removed from member private final String id;

**Comment by elucash** [comment]
Yes, I've reproduced the same. Just released 2.3.7 where field ignores are not used by default, only as a feature flag

**Comment by auke-** [comment]
Thanks!
